docs/topics/sealed-classes.md

```diff
@@ -1,99 +1,350 @@
+    is Error.FileReadError -> println("Error while reading file ${e.file}")
+    is Error.DatabaseError -> println("Error while reading from database ${e.source}")
+    Error.RuntimeError -> println("Runtime error")
+    // No `else` clause is required because all the cases are covered
+}
+//sampleEnd
+
+// List all errors
+fun main() {
+    val errors = listOf(
+        Error.FileReadError("example.txt"),
+        Error.DatabaseError("usersDatabase"),
+        Error.RuntimeError
+    )
+
+    errors.forEach { log(it) }
 }
 ```
+{kotlin-runnable="true" kotlin-min-compiler-version="1.5"}
 
-> `when` expressions on [`expect`](multiplatform-expect-actual.md) sealed classes in the common code of multiplatform projects still 
-> require an `else` branch. This happens because subclasses of `actual` platform implementations aren't known in the 
-> common code.
+> In multiplatform projects, if you have a sealed class with a `when` expression as an 
+> [expected declaration](multiplatform-expect-actual.md) in your common code, you still need an `else` branch. 
+> This is because subclasses of `actual` platform implementations may extend sealed classes that 
+> aren't known in the common code.
 >
 {type="note"}
+
+## Use case scenarios
+
+Let's explore some practical scenarios where the application of sealed classes or interfaces can be particularly helpful.
+
+### State management in UI applications
+
+You can use sealed classes to represent different states of a UI in an application. 
+This approach allows for structured and safe handling of UI changes. 
+This example demonstrates how to manage various UI states:
+
+```kotlin
+sealed class UIState {
+	data object Loading : UIState()
+	data class Success(val data: String) : UIState()
+	data class Error(val exception: Exception) : UIState()
+}
+
+fun updateUI(state: UiState) {
+	when (state) {
+    	is UIState.Loading -> showLoadingIndicator()
+    	is UIState.Success -> showData(state.data)
+    	is UIState.Error -> showError(state.exception)
+	}
+}
+```
+
+### Payment method handling
+
+In practical business applications, handling various payment methods efficiently is a common requirement. 
+This example illustrates how to implement such business logic using sealed classes. 
+By representing different payment methods as subclasses of a sealed class, it establishes a clear and manageable 
+structure for processing transactions:
+
+```kotlin
+sealed class Payment {
+	data class CreditCard(val number: String, val expiryDate: String) : Payment()
+	data class PayPal(val email: String) : Payment()
+	data object Cash : Payment()
+}
+
+fun processPayment(payment: Payment) {
+	when (payment) {
+    	is Payment.CreditCard -> processCreditCardPayment(payment.number, payment.expiryDate)
+    	is Payment.PayPal -> processPayPalPayment(payment.email)
+    	is Payment.Cash -> processCashPayment()
+	}
+}
+```
+
+`Payment` is a sealed class that represents different payment methods in an e-commerce system: 
```

Comment:
Maybe organize these senteces into a list?

* `Payment` ...
* `processPayment()` ...

WDYT?

Reply:
hmm 🤔 I think adding a line break between the processPayment() sentence and the previous ones would aid with readability a bit more. 

(Converting this to a list would look something like this (I think it looks a bit awkward):

* `Payment`: a sealed class that represents different payment methods in an e-commerce system:
  * ˙CreditCard˙
  * ˙Paypal˙
  * ˙Cash˙
* ˙processPayment()˙: a function that demonstrates how to handle different payment methods.

Reply:
WFM!